client/my-sites/domains/components/domain-form-fieldsets/region-address-fieldsets.jsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * External	 dependencies
+ *
+ * @format
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import identity from 'lodash/identity';
+import noop from 'lodash/noop';
+import includes from 'lodash/includes';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CHECKOUT_EU_ADDRESS_FORMAT_COUNTRY_CODES,
+	CHECKOUT_UK_ADDRESS_FORMAT_COUNTRY_CODES,
+} from './constants';
+import UsAddressFieldset from './us-address-fieldset';
+import EuAddressFieldset from './eu-address-fieldset';
+import UkAddressFieldset from './uk-address-fieldset';
+import { Input, HiddenInput } from 'my-sites/domains/components/form';
+
+export class RegionAddressFieldsets extends Component {
+	static propTypes = {
+		getFieldProps: PropTypes.func,
+		translate: PropTypes.func,
+		countryCode: PropTypes.string,
+		canAutoFocusAddressField: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		getFieldProps: noop,
+		translate: identity,
+		countryCode: 'US',
+		canAutoFocusAddressField: false,
+	};
+
+	getAddressInputRefCallback = input => {
```

Comment:
This is the actual callback, so the `get` is a bit misleading, and we can get  We can get address & inputRef from the calling context, so I'd even suggest something like `maybeFocus`?

another option would be to unconditionally focus ( e.g. `invokeFocus = _.method( 'focus' )`, and then do something like `ref={ canAutoFocusAddressField ? method('focus') : noop }`


It's good that we're caching it here so we don't recreate it every render 👍

Reply:
I like the look of [_.method](https://lodash.com/docs/4.17.4#method).  My first instinct is that it might look a little bit clever, since it’s not immediately clear that we’re acting on an argument pass it to by the callback, but the docs are clear and the new import is less code the the internal callback :)